data/transaction/transaction_test.go

```diff
@@ -202,30 +214,76 @@ func TestTransaction_HasOptionGuardianSet(t *testing.T) {
 func TestTransaction_GetDataForSigningShouldWork(t *testing.T) {
 	t.Parallel()
 
-	tx := &transaction.Transaction{}
+	t.Run("without hash sign option set", func(t *testing.T) {
+		t.Parallel()
 
-	numEncodeCalled := 0
-	marshalizerWasCalled := false
-	buff, err := tx.GetDataForSigning(
-		&mock.PubkeyConverterStub{
-			EncodeCalled: func(pkBytes []byte) string {
-				numEncodeCalled++
-				return ""
+		tx := &transaction.Transaction{}
+
+		numEncodeCalled := 0
+		marshalizerWasCalled := false
+		buff, err := tx.GetDataForSigning(
+			&mock.PubkeyConverterStub{
+				EncodeCalled: func(pkBytes []byte) string {
+					numEncodeCalled++
+					return ""
+				},
 			},
-		},
-		&mock.MarshalizerStub{
-			MarshalCalled: func(obj interface{}) (bytes []byte, err error) {
-				marshalizerWasCalled = true
+			&mock.MarshalizerStub{
+				MarshalCalled: func(obj interface{}) (bytes []byte, err error) {
+					marshalizerWasCalled = true
 
-				return make([]byte, 0), nil
+					return make([]byte, 0), nil
+				},
 			},
-		},
-	)
+			&mock.HasherMock{},
```

Comment:
can you check that hasher.Compute not called here?

Reply:
added additional check